libkbfs/block_retrieval_queue.go

```diff
@@ -173,7 +173,7 @@ func (brq *blockRetrievalQueue) Request(ctx context.Context, priority int, kmd K
 			// specific type where the request blocks are CommonBlocks, but
 			// that direction can Set correctly. The cache will never have
 			// CommonBlocks.
-			cachedBlock, hasPrefetched, err :=
```

Comment:
In `FinishSync` we `DeletePermanent` for each pointer in `syncState.newIndirectFileBlockPtrs`, but I don't see where we insert those items into the transient cache. Is that done somewhere else?

The race I'm worried about is what happens if `GetWithPrefetch` returns a `PermanentEntry`, but it's deleted before you can `Put` it back.